### DISCUSSION TITLE
Text Color Change for different fonts

### DISCUSSION THREAD

**Discussion by meghanaviyyapu** [pymupdf/PyMuPDF#3482] [Looking for help]
I have tried modifying the font color using the attached code but could not view any changes in the text color and could see only rectangles drawn.
Attached the code 
[fontcolos.txt](https://github.com/pymupdf/PyMuPDF/files/15320880/fontcolos.txt)
[color.pdf](https://github.com/pymupdf/PyMuPDF/files/15320881/color.pdf)
and the output PDF. Could you let me know if I missed something?

**Comment by JorjMcKie** [comment]
You said you want to change the color of the text only. Not the text itself.
And you are trying to insert a different text???

**Comment by meghanaviyyapu** [comment]
Yes to change the color of the first line of text in the PDF, I tried inserting the same text present in the first line of PDF with a different font color

**Comment by JorjMcKie** [comment]
Anyway, I checked out all of the embedded fonts. They all are font subsets.
The result: not a single one of these font subsets is built in a way such that it can be reused for text insertion.
So you either have to give up the whole idea, or you have to find the original fonts for each of these subsets and insert the text using those.

**Comment by meghanaviyyapu** [comment]
Thanks for checking. Could you let me know how the original fonts could be found for the font subsets?

**Reply by JorjMcKie** [reply]
well, at the normal places. Best search for them via Google "font Mincho" etc.

**Reply by JorjMcKie** [reply]
The Times Roman is no problem, simply use the Base14 font.

**Reply by meghanaviyyapu** [reply]
Is it possible to just change the color without changing text?

**Reply by meghanaviyyapu** [reply]
I need to insert text using the same font extracted via span["font"] with a different font color. For the above PDF, is there any way to modify using the extracted font and not Base14 fonts?

**Reply by JorjMcKie** [reply]
No - I think I answered that in some other thread already. There is no reliable way to do that.

**Comment by JorjMcKie** [comment]
> I need to insert text using the same font extracted via span["font"] with a different font color. For the above PDF, is there any way to modify using the extracted font and not Base14 fonts?

You can choose any variant of Times-Roman you like.

**Reply by meghanaviyyapu** [reply]
Ok but the font should be the same as the font extracted via PyMuPDF

**Comment by JorjMcKie** [comment]
Which color do you want instead of that violet-like one?

**Reply by meghanaviyyapu** [reply]
Any color is fine but the font should be the same as the one present in the input PDF

**Reply by JorjMcKie** [reply]
Don't understand - the **_same_** color? What do you want to change then?

**Reply by meghanaviyyapu** [reply]
No different color (black) but with the same font properties extracted via PyMuPDF

**Comment by meghanaviyyapu** [comment]
Can you please let me know how color for the text in the attached PDF can be changed? I have tried changing the color using page.insert_text() but could not view the text after conversion. Could you please let me know the reason?
[Chine.pdf](https://github.com/pymupdf/PyMuPDF/files/15413054/Chine.pdf)
